tests/README.md

```diff
@@ -1,25 +1,239 @@
+```
+
+#### Activate the Virtual Environment
+
+```bash
+$ source <name-of-virtualenv-folder>/bin/activate
+```
+
+This now means that you have a clean python environment, and no packages you
+have previously installed outside of this virtual environment will be
+discoverable by Python.
+
+#### Install the Python Dependencies from the Requirements File
+
+```bash
+$ pip install -r requirements/integration/python-requirements.txt
+```
+
+This will install the requirements needed to run the tests.
+
+### Running the Tests
+
+Next, run the tests with the `run.sh` script.
+
+```bash
+$ ./run.sh
+```
+
+**NOTE**: This is dependent upon having a functioning Docker environment, and being
+logged in to `registry.mender.io`.
+
+## Modifying the Docker Images Employed
+
+In order to run the integration tests with the local changes made to some Mender
+service, it is necessary to build the container, and tag it with the matching
+tag employed in the integration repository. This can be found in the
+`docker-compose.yml` file in the root directory, under the `image:` key.
+
+#### Example -- Running with a Custom Backend Service
+
+For the backend services there exists a `Dockerfile` in the root repository, and
+as such a new image can be built and tagged by
+
+```bash
+$ cd /path/to/<service>/
+$ docker build . -t mendersoftware/<service>:master
+```
+
+#### Example -- Running with a Custom Client
+
+For building a custom client the approach is a little different, due to the fact
+that the client comes bundles with a Yocto image. Therefore, in order to build
+and run a custom client with the integration test setup, first build a Yocto
+image, containing the custom client. Then, build a Docker image containing this
+client by
+
+```bash
+$ cd /path/to/meta-mender
+$ cd meta-mender-qemu/docker
+$ ./build-docker qemux86-64 -t mendersoftware/mender-client-qemu:master
+```
+
+This will error if the Yocto environment has not been sourced yet.
+
+
+-------------------------------------------------------------------------------
+
+
+## Running the Tests in a VM
+
+### Vagrant
+
+The setup provides a
+[Vagrantfile](https://www.vagrantup.com/intro/getting-started/) in order to
+provide a common ground for running the integration tests on, based on an
+Ubuntu18.04 image. This can be found in `tests/vagrant`, and provisions a VM
+with the correct setup. The Vagrantfile will provision a fresh Ubuntu18.04
+image, and install all the *Debian* and *Python* requirements in the
+*requirements/(deb|python)requirements.txt* file.
+
+This requires that Vagrant is installed, as well as having the
+
+| Environment variable      | Reason |
+| ---                       | ---    |
+|GOPATH                     | Used for mounting your local GO tree in the VM as a synced folder |
+|DOCKER_USER                | Needed for logging in to the Mender docker registry  |
+|MENDER_DOCKER_REGISTRY_PWD |  Needed for logging in to the Mender docker registry  |
```

Comment:
Maybe sync the naming for these two: `MENDER_DOCKER_REGISTRY_USER` and `MENDER_DOCKER_REGISTRY_PWD`